### PR TITLE
Support outputting result in s-expression or debug form

### LINKED ISSUE #96: Add S-expression debug formatter for ASTs
We used to have this as the default formatter but got rid of it to make easier pretty-printing. let's add it back.

### LINKED ISSUE #97: Add CLI parser
So that we can pass and handle extra args (for example, stop after parsing or show S-expression form after #96 lands). [clap](https://github.com/clap-rs/clap) is probably the best way to go here.

### PR DESCRIPTION
**Pull request by ayazhafiz**
Also adds an option to dump the AST after parsing.

Closes #96
Closes #97

### CODE REVIEW

**Comment by ayazhafiz** [comment]
Relevant to #92:

```bash
❯ cargo run "a+1-1" -os-expression
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/slide a+1-1 -os-expression`
(- (+ a 1) 1)
```

**Comment by luke-bhan** [comment]
im slightly confused. How do we currently handle this expression in non-debug form?

File: slide/src/main.rs [hunk 1]
```diff
@@ -1,24 +1,83 @@
...
+
+fn print<T>(obj: T, output_form: OutputForm) -> String
+where
+    T: Grammar,
+{
+    match output_form {
+        OutputForm::Pretty => obj.to_string(),
+        OutputForm::SExpression => obj.s_form(),
+        OutputForm::Debug => format!("{:#?}", obj),
+    }
```
Comment (luke-bhan) [inline slide/src/main.rs:82]: how is the debug form defined. By obj format implementation which is the same as to_string() right? So why do we have it?

**Comment by luke-bhan** [comment]
other then the debug form, LGTM

**Comment by ayazhafiz** [comment]
Debug form looks like this:

```rust
❯ cargo run "a+1-1" -odebug
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/slide a+1-1 -odebug`
BinaryExpr(
    BinaryExpr {
        op: Minus,
        lhs: BinaryExpr(
            BinaryExpr {
                op: Plus,
                lhs: Var(
                    "a",
                ),
                rhs: Const(
                    1.0,
                ),
            },
        ),
        rhs: Const(
            1.0,
        ),
    },
)
```

**Pull request merged**